specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,203 @@
+
+    if (result == 
+        COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT::COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_HTCAPTION)
+    {
+        return HTCAPTION;
+    }
+
+    break;
+    ....
+}
+....
+}
+```
+#### C#/WinRT
+```c#
+protected override void WndProc(ref Message m)
+{
+....
+
+bool isCapturingMouse = false;
+
+if (m.Msg == WM_NCHITTEST)
+{
+    Point point = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+    if (CompositionController.NonClientHitTestAtPoint(point) == 
+        CoreWebView2NonClientHitTestResult.Htcaption) 
+    {
+        m.Result = (IntPtr)HTCAPTION;
+    }
+}
+// The following code forwards mouse messages to the WebView2 - Start
+else if (m.msg >= WM_MOUSEFIRST && m.msg <= WM_MOUSELAST) 
+{
+  
+    Point point = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+    // IsPointInWebview is an example of a function which your app can implement that checks if 
+    // the mouse message point lies inside the rect of the visual that hosts the WebView
+    if (IsPointInWebView(point) || m.msg == WM_MOUSELEAVE || isCapturingMouse) 
+    {
+
+        if (m.msg == WM_NCRBUTTONDOWN && m.w_param == HTCAPTION) 
+        {
+            // capturing the mouse will allow us to begin forwarding messages 
+            // to webview
+            isCapturingMouse = true;
+            SetCapture(mainWindowHwnd);
+        }
+        else if (message ==  WM_RBUTTONUP && GetCapture() == mainWindowHwnd) 
+        {
+            // no longer need to capture the mouse for draggbale regions support 
+            // after this
+            isCapturingMouse = false;
+            ReleaseCapture();
+        }
+        
+        // forward mouse messages to WV
+        CompositionController.SendMouseInput(
+            (CoreWebView2MouseEventKind)message,
+            (CoreWebView2MouseEventVirtualKeys)GET_KEYSTATE_WPARAM(wParam),
+            mouseData,
+            point);
+
+        m.Result = 0
+        return;
+    }
+        
+}
+// The following code forwards mouse messages to the WebView2 - Start
+    ....
+}
+       
+```
+# API Details
+## Win32 C++
+```cpp
+/// This enum contains values representing possible regions that Webview
+/// can support
+typedef enum COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT {
+    /// Caption region
+    COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_CAPTION,
+    /// Client region
+    COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_CLIENT,
+    /// Out of bounds of the app window
+    COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_NONE
+} COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT;
+
+/// This interface includes the new API for enabling Webview support for hit-testing regions
+[uuid(42BF7BA5-917A-4C27-ADA1-EA6969854C16), object, pointer_default(unique)]
+interface ICoreWebView2StagingCompositionController4 : ICoreWebView2CompositionController3 {
+    /// Takes in a point and returns a Hit-test result value as an out parameter
+    /// point is in Screen Coordinates
```

Comment:
Additionally in these reference docs it would be good to say when it returns the different hit test enum values. Seems like it always returns client unless its over an HTML element using the app region CSS style (or whatever that's called?). It would be good to link to public documentation on MDN (https://developer.mozilla.org/en-US/) about the app region CSS (assuming MDN has docs on it). And you should also describe what happens when the hit test is out of bounds of the WebView2.

Reply:
I can't find any docs on MDN for app-region. Also, hit testing a point out of bounds of webview is what COREWEBVIEW2_HIT_TEST_RESULT_NONE is for. There was some contemplation about calling it COREWEBVIEW2_HIT_TEST_RESULT_OUT_OF_BOUNDS instead. What do you think?